### PR TITLE
lun-UID2-1426-gcp-oidc-attestation-client add OIDC provider

### PR DESCRIPTION
**Pull request by lunwang-ttd**

### CODE REVIEW

File: src/main/java/com/uid2/attestation/gcp/OidcAttestationProvider.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+import com.uid2.enclave.IAttestationProvider;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+
+public class OidcAttestationProvider implements IAttestationProvider {
+	// the local file contains OIDC token
+	private String tokenFilePath = "/run/container_launcher/attestation_verifier_claims_token";
```
Comment (yishi-ttd) [inline src/main/java/com/uid2/attestation/gcp/OidcAttestationProvider.java]: Shoud we put this tokenFilePath to ctor, define a const default path, and provide an additional no-arg ctor use this default value?

File: src/main/java/com/uid2/attestation/gcp/OidcAttestationProvider.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+	@Override
+	public byte[] getAttestationRequest(byte[] publicKey) throws AttestationException {
```
Comment (thomasm-ttd) [inline src/main/java/com/uid2/attestation/gcp/OidcAttestationProvider.java]: Is this work in progress? I assume publickKey will be used later?

Comment (lunwang-ttd) [reply src/main/java/com/uid2/attestation/gcp/OidcAttestationProvider.java]: It's completed, OIDC doesn't need public key

Comment (lunwang-ttd) [reply src/main/java/com/uid2/attestation/gcp/OidcAttestationProvider.java]: I added a SuppressWarning("unused") annotation to make it clearer

File: src/main/java/com/uid2/attestation/gcp/OidcAttestationProvider.java (hunk 1 above)
Comment (lunwang-ttd) [reply src/main/java/com/uid2/attestation/gcp/OidcAttestationProvider.java]: 👍 updated

File: src/main/java/com/uid2/attestation/gcp/OidcAttestationProvider.java [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+import com.uid2.enclave.IAttestationProvider;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+
+public class OidcAttestationProvider implements IAttestationProvider {
+	// the local file contains OIDC token
+	private String tokenFilePath;
```
Comment (yishi-ttd) [inline src/main/java/com/uid2/attestation/gcp/OidcAttestationProvider.java]: NIT: this could be marked as final.

**Review by yishi-ttd — APPROVED** [review]

File: src/main/java/com/uid2/attestation/gcp/OidcAttestationProvider.java (hunk 3 above)
Comment (lunwang-ttd) [reply src/main/java/com/uid2/attestation/gcp/OidcAttestationProvider.java]: Fixed

**Pull request merged**